rust/cymbal/src/types/mod.rs

```diff
@@ -106,6 +113,40 @@ impl ErrProps {
             serde_json::Value::Array(errors),
         );
     }
+
+    pub fn to_fingerprinted(self, fingerprint: String) -> FingerprintedErrProps {
+        FingerprintedErrProps {
+            exception_list: self.exception_list,
+            fingerprint,
+            other: self.other,
+        }
+    }
+}
+
+impl FingerprintedErrProps {
+    pub fn generate_new_issue_name(&self) -> String {
+        if let Some(exception) = self.exception_list.first() {
+            // We just use the first exception in the list
+            format!(
+                "{}: {}",
+                exception.exception_type, exception.exception_message
+            )
+        } else {
+            // We don't have a type-system way to assert that this is unreachable (or rather,
+            // to encode this in the type system would be cumbersome), but it if it's ever reached,
+            // we've got a bug.
+            unreachable!("Tried to generate issue name with no exception in exception list")
+        }
+    }
```

Comment:
Currently on the frontend we set the name to be the `exception_type` and the description to be `exception_message`. I think it could be worth doing that in this case too